core/internal/gethwrappers/go_generate_test.go

```diff
@@ -1,70 +1,59 @@
 // package gethwrappers_test verifies correct and up-to-date generation of golang wrappers
 // for solidity contracts. See go_generate.go for the actual generation.
-package gethwrappers_test
+package gethwrappers
 
 import (
-	"bufio"
 	"crypto/sha256"
-	"encoding/json"
 	"fmt"
-	"io"
 	"io/ioutil"
 	"os"
 	"os/exec"
-	"path"
 	"path/filepath"
-	"regexp"
 	"strings"
 	"testing"
 
 	"github.com/smartcontractkit/chainlink/core/utils"
 
 	gethParams "github.com/ethereum/go-ethereum/params"
-	"github.com/pkg/errors"
-	"github.com/tidwall/gjson"
 
 	"github.com/fatih/color"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-// contractVersion records information about the solidity compiler artifact a
-// golang contract wrapper package depends on.
-type contractVersion struct {
-	// path to compiler artifact used by generate.sh to create wrapper package
-	compilerArtifactPath string
-	// hash of the artifact at the timem the wrapper was last generated
-	hash string
-}
-
-// integratedVersion carries the full versioning information checked in this test
-type integratedVersion struct {
-	// Version of geth last used to generate the wrappers
-	gethVersion string
-	// { golang-pkg-name: version_info }
-	contractVersions map[string]contractVersion
-}
-
 // TestCheckContractHashesFromLastGoGenerate compares the metadata recorded by
 // record_versions.sh, and fails if it indicates that the corresponding golang
 // wrappers are out of date with respect to the solidty contracts they wrap. See
 // record_versions.sh for description of file format.
 func TestCheckContractHashesFromLastGoGenerate(t *testing.T) {
-	versions := readVersionsDB(t)
-	require.NotEmpty(t, versions.gethVersion,
+	versions, err := ReadVersionsDB()
+	require.NoError(t, err)
+	require.NotEmpty(t, versions.GethVersion,
 		`version DB should have a "GETH_VERSION:" line`)
-	require.Equal(t, versions.gethVersion, gethParams.Version,
-		color.HiRedString(boxOutput("please re-run `go generate ./core/services/vrf` and commit the changes")))
-	for _, contractVersionInfo := range versions.contractVersions {
+	wd, err := os.Getwd()
+	if err != nil {
+		wd = "<directory containing this test>"
+	}
+	require.Equal(t, versions.GethVersion, gethParams.Version,
+		color.HiRedString(boxOutput("please re-run `go generate %s` and commit the"+
+			"changes", wd)))
+	for _, contractVersionInfo := range versions.ContractVersions {
 		compareCurrentCompilerAritfactAgainstRecordsAndSoliditySources(
 			t, contractVersionInfo)
 	}
+	// Just check that LinkToken details haven't changed (they never ought to)
+	linkDetails, err := ioutil.ReadFile(filepath.Join(getProjectRoot(t),
+		"evm-test-helpers/src/LinkToken.json"))
+	require.NoError(t, err, "could not read link contract details")
+	require.Equal(t, fmt.Sprintf("%x", sha256.Sum256(linkDetails)),
```

Comment:
I use simpler logic here: Just the hash of the whole JSON file, rather than of the extracted, normalized ABI and the binary object. If `LinkToken.json` ever changes in any way, this test will complain, but it very likely wont.